src/plugins/intel_cpu/src/graph.cpp

```diff
@@ -1315,14 +1348,83 @@ void Graph::InferDynamic(SyncInferRequest* request) {
 }
 
 inline void Graph::ExecuteNode(const NodePtr& node, const dnnl::stream& stream) const {
-    DUMP(node, getConfig().debugCaps, infer_count);
-
-    OV_ITT_SCOPED_TASK(itt::domains::intel_cpu, node->profiling.execute);
-    DEBUG_LOG(*node);
-    if (node->isDynamicNode()) {
-        node->executeDynamic(stream);
+    if (!node->parallelWith.empty()) {
+        // run nodes in parallel
+        auto& parallelNodes = node->parallelWith;
+        if (node == parallelNodes[0]) {
+            auto cpuExecutor =
+                std::dynamic_pointer_cast<ov::threading::CPUStreamsExecutor>(context->getStreamExecutor());
+
+            if (cpuExecutor) {
+                auto num_parallel_nodes = parallelNodes.size();
+                ParalleMtNuma(num_parallel_nodes, cpuExecutor, [&](int subStreamID, size_t i) {
+                    auto& n = parallelNodes[i];
+
+                    n->toNumaNode(subStreamID);
+                    if (n->isDynamicNode()) {
+                        n->executeDynamic(stream);
+                    } else {
+                        n->execute(stream);
+                    }
+                });
+            } else {
+                // fallback to serialize executor
+                for (auto& node : parallelNodes) {
+                    if (node->isDynamicNode()) {
+                        node->executeDynamic(stream);
+                    } else {
+                        node->execute(stream);
+                    }
+                }
+            }
+        }
     } else {
-        node->execute(stream);
+        DUMP(node, getConfig().debugCaps, infer_count);
+        OV_ITT_SCOPED_TASK(itt::domains::intel_cpu, node->profiling.execute);
+        DEBUG_LOG(*node);
+        if ((getGraphContext()->getStreamExecutor()) && (getConfig().hintPerfMode == ov::hint::PerformanceMode::LATENCY)) {
+            node->toNumaNode(getGraphContext()->getStreamExecutor()->get_numa_node_id());
+        }
```

Comment:
May the numa node be changed from infer request to infer request? If not, it's better to call this action once during compilation.
If yes, at least the condition can be evaluated once and stored in the graph.

Reply:
> May the numa node be changed from infer request to infer request? If not, it's better to call this action once during compilation. If yes, at least the condition can be evaluated once and stored in the graph.

Agree with you.
This is a temp for SD model. From perf perspective, the latency gains and tput is no-drop.
Plan to remove this hardcode and try to allocate on numa node at lower level memory management.